File: codechef/JUNE17/unionset.cpp
#include<bits/stdc++.h>
#define fn(i,n) for(int i=0;i<n;i++)
#define wh while
#define ll long long
#define sc(n) scanf("%lld",&n)
#define mx 12500
using namespace std;
int main(){
	ll t;
	cin>>t;
	while (t--){
		ll N,K,count2,total;
		sc(N);sc(K);
		bool lookup[K+1];
		fn(i,K+1){
		lookup[i]=false;
		}
		ll all_list[N][K+1];
		fn(i,N){
			sc(all_list[i][0]);
			fn(j,all_list[i][0]){
				sc(all_list[i][j+1]);
			}
		}	
	total=0;
	fn(i,N){
		ll count1=0;
		fn (m,all_list[i][0]){
			lookup[all_list[i][m+1]]=true;
		}
count1=all_list[i][0];
ll count2=0;
for (int j=i+1;j<N;j++){
count2=0;
if (count1+all_list[j][0]>=K){
	fn(m,all_list[j][0]){
	  if (lookup[all_list[j][m+1]]==false)
			count2=count2+1;
	}
	if (count1+count2==K)
		total+=1;
}
}
fn(m,all_list[i][0])
	lookup[all_list[i][m+1]]=false;
}
cout<<total<<endl;
}
return 0;
}

File: codechef/JUNE17/unionset.py
t=input()
while t>0:
	t-=1
	N,K=map(int,raw_input().split())
	#lookup=[[0 for x in range(K)] for y in range(N)]
	all_list=[]
	lookup=[False]*(K+1)
	for i in range(N):
		arr=map(int,raw_input().split())
		#min_=min(min_,arr[0])
		all_list.append(arr)
	total=0
	for i in range(N):
		count1=0
		for m in range(all_list[i][0]):
			lookup[all_list[i][m+1]]=True
		count1=all_list[i][0]
		count2=0
		for j in range(i+1,N):
			count2=0
			if count1+all_list[j][0]>=K:
				for m in range(all_list[j][0]):
					if lookup[all_list[j][m+1]]==False:
						count2+=1
				if count1+count2==K:
					total+=1
		for m in range(all_list[i][0]):
			lookup[all_list[i][m+1]]=False
	print total	
